src/main/java/frc/robot/subsystems/climber/ClimberIOTalonFX.java

```diff
@@ -0,0 +1,268 @@
+package frc.robot.subsystems.climber;
+
+import static frc.robot.subsystems.climber.ClimberConstants.*;
+
+import com.ctre.phoenix6.BaseStatusSignal;
+import com.ctre.phoenix6.StatusCode;
+import com.ctre.phoenix6.StatusSignal;
+import com.ctre.phoenix6.configs.CurrentLimitsConfigs;
+import com.ctre.phoenix6.configs.TalonFXConfiguration;
+import com.ctre.phoenix6.controls.PositionTorqueCurrentFOC;
+import com.ctre.phoenix6.controls.PositionVoltage;
+import com.ctre.phoenix6.controls.TorqueCurrentFOC;
+import com.ctre.phoenix6.controls.VelocityTorqueCurrentFOC;
+import com.ctre.phoenix6.controls.VoltageOut;
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.ctre.phoenix6.signals.InvertedValue;
+import com.ctre.phoenix6.signals.NeutralModeValue;
+import frc.lib.team3015.subsystem.FaultReporter;
+import frc.lib.team3061.RobotConfig;
+import frc.lib.team3061.drivetrain.swerve.Conversions;
+import frc.lib.team6328.util.Alert;
+import frc.lib.team6328.util.Alert.AlertType;
+import frc.lib.team6328.util.TunableNumber;
+
+/** TalonFX implementation of the generic SubsystemIO */
+public class ClimberIOTalonFX implements ClimberIO {
+  private TalonFX leftMotor;
+  private TalonFX rightMotor;
+
+  private PositionTorqueCurrentFOC leftPositionCurrentRequest;
+  private VelocityTorqueCurrentFOC leftVelocityRequest;
+
+  private PositionTorqueCurrentFOC rightPositionCurrentRequest;
+  private VelocityTorqueCurrentFOC rightVelocityRequest;
+
+  private double leftRequestedVelocity;
+  private double leftRequestedSetpoint; 
+
+  private double rightRequestedVelocity;
+  private double rightRequestedSetpoint;
+
+  private StatusSignal<Double> leftVelocityRPMStatusSignal;
+  private StatusSignal<Double> leftPositionStatusSignal;
+  private StatusSignal<Double> leftStatorCurrentAmpsStatusSignal;
+  private StatusSignal<Double> leftSetpointStatusSignal;
+  private StatusSignal<Double> leftVoltageRequest;
+
+  private StatusSignal<Double> rightVelocityRPMStatusSignal;
+  private StatusSignal<Double> rightPositionStatusSignal;
+  private StatusSignal<Double> rightStatorCurrentAmpsStatusSignal;
+  private StatusSignal<Double> rightSetpointStatusSignal;
+
+  private Alert configAlert =
+      new Alert("Failed to apply configuration for Climber.", AlertType.ERROR);
+
+  private final TunableNumber leftMotorKP = new TunableNumber("Climber/leftMotorKP", LEFT_POSITION_PID_P);
+  private final TunableNumber leftMotorKI = new TunableNumber("Climber/leftMotorKI", LEFT_POSITION_PID_I);
+  private final TunableNumber leftMotorKD = new TunableNumber("Climber/leftMotorKD", LEFT_POSITION_PID_D);
+  private final TunableNumber leftMotorKS =
```

Comment:
add more tunable based on the additional constants specified in ClimberConstants